samples/subsys/testsuite/pytest/shell/pytest/test_shell.py

```diff
@@ -22,15 +21,25 @@ def wait_for_message(iter_stdout, message, timeout=60):
             return False
 
 
-def test_shell_print_help(dut: DeviceAbstract):
-    time.sleep(1)  # wait for application initialization on DUT
+def wait_for_prompt(dut: DeviceAbstract, prompt='uart:~$', timeout=20):
+    time_started = time.time()
+    while True:
+        dut.write(b'\n')
+        for line in dut.iter_stdout:
```

Comment:
How is this guaranteed that it won't block forever?

Reply:
It shouldn't block. `iter_stdout` calls `readline` method, that is blocking, but in device adapters there is implementation to avoid blocking forever: in `qemu_adapter` and `simulator_adapter` (for native_posix) are threads and queues used to set timeout. In `hardware_adapter` the timeout is passed during Serial initialization.

Reply:
Unfortunately, when using Serial (for hardware) the `readline` is blocking. We must change this helper method. Thanks @pdgendt for comment